relayer/relays/beacon/header/header.go

```diff
@@ -319,7 +352,42 @@ func (h *Header) populateClosestCheckpoint(slot uint64) (cache.Proof, error) {
 	return checkpoint, nil
 }
 
+func (h *Header) populateCheckPointCacheWithDataFromChain(slot, checkpointSlot uint64) (uint64, error) {
+	log.WithFields(log.Fields{"calculatedCheckpointSlot": checkpointSlot}).Info("checkpoint slot not available, try with slot in next sync period instead")
+
+	checkpointSlot = h.syncer.CalculateNextCheckpointSlot(slot)
+
+	lastFinalizedHeaderState, err := h.writer.GetLastFinalizedHeaderState()
+	if err != nil {
+		return 0, fmt.Errorf("get last finalized header for the checkpoint: %w", err)
+	}
+
+	if slot > lastFinalizedHeaderState.BeaconSlot {
+		return 0, ErrBeaconHeaderNotFinalized
+	}
+
+	if checkpointSlot < lastFinalizedHeaderState.BeaconSlot {
+		historicState, err := h.findClosestCheckPoint(slot)
+		if err != nil {
+			return 0, fmt.Errorf("get history finalized header for the checkpoint: %w", err)
+		}
+		checkpointSlot = historicState.BeaconSlot
+	} else {
+		// Setting the checkpoint slot to what is the latest finalized header on-chain, since the checkpoint should
+		// not be after the latest finalized header on-chain
+		checkpointSlot = lastFinalizedHeaderState.BeaconSlot
+	}
```

Comment:
Seems if-else branch duplicated here.

Reply:
I don't understand, apologies?

Reply:
Sorry, I'm looking in the wrong. Just ignore it!